dali/aux/optical_flow/turing_of/optical_flow_turing.cc

```diff
@@ -0,0 +1,169 @@
+  refbuf_.reset(nullptr);
+  outbuf_.reset(nullptr);
+  TURING_OF_API_CALL(turing_of_.nvOFDestroy(of_handle_));
+}
+
+
+void OpticalFlowTuring::CalcOpticalFlow(
+        dali::kernels::TensorView<dali::kernels::StorageGPU, const uint8_t, 3> reference_image,
+        dali::kernels::TensorView<dali::kernels::StorageGPU, const uint8_t, 3> input_image,
+        dali::kernels::TensorView<dali::kernels::StorageGPU, float, 3> output_image,
+        dali::kernels::TensorView<dali::kernels::StorageGPU, const float, 3> external_hints) {
+  kernel::RgbToRgba(input_image.data, reinterpret_cast<uint8_t *>(inbuf_->GetPtr()),
+                    inbuf_->GetStride().x, width_, height_);
+  kernel::RgbToRgba(reference_image.data, reinterpret_cast<uint8_t *>(refbuf_->GetPtr()),
+                    refbuf_->GetStride().x, width_, height_);
+  cudaStreamSynchronize(stream_);
+
+  auto in_params = GenerateExecuteInParams(inbuf_->GetHandle(), refbuf_->GetHandle());
+  auto out_params = GenerateExecuteOutParams(outbuf_->GetHandle());
+  TURING_OF_API_CALL(turing_of_.nvOFExecute(of_handle_, &in_params, &out_params));
+  cudaStreamSynchronize(stream_);
+
+  kernel::DecodeFlowComponents(reinterpret_cast<int16_t *>(outbuf_->GetPtr()), output_image.data,
+                               outbuf_->GetStride().x, outbuf_->GetDescriptor().width,
+                               outbuf_->GetDescriptor().height);
+}
+
+
+void OpticalFlowTuring::SetInitParams(dali::optical_flow::OpticalFlowParams api_params) {
+  of_params_.width = static_cast<uint32_t>(width_);
+  of_params_.height = static_cast<uint32_t>(height_);
+
+  if (api_params.grid_size == VectorGridSize::SIZE_4) {
+    of_params_.outGridSize = NV_OF_OUTPUT_VECTOR_GRID_SIZE_4;
+    of_params_.hintGridSize = NV_OF_HINT_VECTOR_GRID_SIZE_4;
+  } else {
+    of_params_.outGridSize = NV_OF_OUTPUT_VECTOR_GRID_SIZE_UNDEFINED;
+    of_params_.hintGridSize = NV_OF_HINT_VECTOR_GRID_SIZE_UNDEFINED;
+  }
+
+  of_params_.mode = NV_OF_MODE_OPTICALFLOW;
+
+  if (api_params.perf_quality_factor >= 0.0 && api_params.perf_quality_factor < 0.375f) {
+    of_params_.perfLevel = NV_OF_PERF_LEVEL_SLOW;
+  } else if (api_params.perf_quality_factor < 0.75f) {
+    of_params_.perfLevel = NV_OF_PERF_LEVEL_MEDIUM;
+  } else if (api_params.perf_quality_factor <= 1.0f) {
+    of_params_.perfLevel = NV_OF_PERF_LEVEL_FAST;
+  } else {
+    of_params_.perfLevel = NV_OF_PERF_LEVEL_UNDEFINED;
+  }
+
+  of_params_.enableExternalHints = api_params.enable_hints ? NV_OF_TRUE : NV_OF_FALSE;
+  of_params_.enableOutputCost = NV_OF_FALSE;
+  of_params_.hPrivData = NULL;
+}
+
+
+NV_OF_EXECUTE_INPUT_PARAMS OpticalFlowTuring::GenerateExecuteInParams
+        (NvOFGPUBufferHandle in_handle, NvOFGPUBufferHandle ref_handle) {
+  NV_OF_EXECUTE_INPUT_PARAMS params;
+  params.inputFrame = in_handle;
+  params.referenceFrame = ref_handle;
+  params.disableTemporalHints = NV_OF_TRUE;
+  params.externalHints = nullptr;
+  params.padding = 0;
+  params.hPrivData = NULL;
```

Comment:
Why `NULL` and not `nullptr`?

Reply:
I'll keep it `NULL`, since OF API docs say it should be `NULL`